File: test-suite/src/test/java/io/micronaut/docs/qualifiers/replaces/defaultimpl/CustomResponseStrategy.java
package io.micronaut.docs.qualifiers.replaces.defaultimpl;

//tag::clazz[]
import io.micronaut.context.annotation.Replaces;
import javax.inject.Singleton;

@Singleton
@Replaces(ResponseStrategy.class)
public class CustomResponseStrategy implements ResponseStrategy {

}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/qualifiers/replaces/defaultimpl/CustomResponseStrategy.kt
package io.micronaut.docs.qualifiers.replaces.defaultimpl

//tag::clazz[]
import io.micronaut.context.annotation.Replaces
import javax.inject.Singleton

@Singleton
@Replaces(ResponseStrategy::class)
class CustomResponseStrategy : ResponseStrategy
//end::clazz[]